player/player.go

```diff
@@ -48,47 +50,76 @@ func PlayURI(uri string, rangeHeader string, w http.ResponseWriter) (err error)
 	if err != nil {
 		return err
 	}
-	rs.setRangeFromHeader(rangeHeader)
-	blobStart, blobEnd := rs.getBlobsRange()
 	rs.prepareWriter(w)
-	err = rs.streamBlobs(blobStart, blobEnd, w)
+	http.ServeContent(w, req, "test", time.Time{}, rs)
 	return err
 }
 
-func parseRange(header string) (int64, int64) {
-	r := regexp.MustCompile(`bytes=(\d+)-(\d*)`)
-	m := r.FindStringSubmatch(header)
-	if len(m) == 0 {
-		return 0, 0
-	}
-	start, err1 := strconv.ParseInt(m[1], 10, 64)
-	end, err2 := strconv.ParseInt(m[2], 10, 64)
-	if err1 != nil {
-		start = 0
-	}
-	if err2 != nil {
-		end = 0
-	}
-	if start < 0 {
-		start = 0
-	}
-	if end < 0 {
-		end = 0
-	}
-	if end > 0 && start > 0 && start > end {
-		start = 0
-		end = 0
-	}
-	return start, end
-}
-
 func newReflectedStream(uri string) (rs *reflectedStream, err error) {
 	client := ljsonrpc.NewClient(config.Settings.GetString("Lbrynet"))
 	rs = &reflectedStream{URI: uri}
 	err = rs.resolve(client)
 	return rs, err
 }
 
+// Read implements io.ReadSeeker interface
+func (s *reflectedStream) Read(p []byte) (n int, err error) {
+	var startOffsetInBlob int64
+
+	bufferLen := len(p)
+	seekOffsetEnd := s.seekOffset + int64(bufferLen)
+	startBlob := int(s.seekOffset / (stream.MaxBlobSize - 2))
+	endBlob := int(seekOffsetEnd / (stream.MaxBlobSize - 2))
+
+	if stream.MaxBlobSize > s.seekOffset {
+		startOffsetInBlob = s.seekOffset
+	} else {
+		startOffsetInBlob = s.seekOffset - int64(startBlob*stream.MaxBlobSize) + 1
+	}
+
+	monitor.Logger.WithFields(log.Fields{
+		"read_buffer_length": bufferLen,
+		"start_blob":         startBlob,
+		"end_blob":           endBlob,
+		"current_offset":     s.seekOffset,
+		"offset_in_blob":     startOffsetInBlob,
+	}).Infof("requesting to read bytes %v..%v from blob stream", s.seekOffset, seekOffsetEnd)
+
+	n, err = s.streamBlobs(startBlob, endBlob, startOffsetInBlob, p)
+	s.seekOffset += int64(n)
+
+	return n, err
+}
+
+// Seek implements io.ReadSeeker interface
+func (s *reflectedStream) Seek(offset int64, whence int) (int64, error) {
+	var zero, newSeekOffset int64
+
+	if whence == io.SeekEnd {
+		newSeekOffset = s.Size - 1 - offset
+	} else if whence == io.SeekStart {
+		newSeekOffset = offset
+	} else if whence == io.SeekCurrent {
+		newSeekOffset = s.seekOffset + offset
+	} else {
+		return zero, e.New("invalid seek whence argument")
```

Comment:
why do you need a `zero` var? why not just return `0`?

Reply:
I think it was a brain fart